### PR TITLE
feat(StyleProvider): add StyleProvider handle cssinjs features

### PR DESCRIPTION
**Pull request by CCherry07**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / document update
- [x] Component style update
- [x] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
增加 StyleProvider  处理 样式兼容 等问题
> 2. Resolve what problem.
> 3. Related issue link.
#6407 
### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
```vue
<template>
    <a-style-provider hash-priority="high">
       <App/>
    </a-style-provider>
</template>
```
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
无
> 2. What will say in changelog?
无
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
custom cssinjs
> 2. Chinese description (optional)
自定义 cssinjs 特性
### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

**Review by tangjinzhou — CHANGES_REQUESTED** [review]

File: components/_util/cssinjs/StyleContext.tsx [hunk 1]
```diff
@@ -71,41 +73,92 @@ export interface StyleContextProps {
...
+
+export const StyleProvider = defineComponent({
+  name: 'AStyleProvider',
+  props: initDefaultProps(AStyleProviderProps(), defaultStyleContext),
+  setup(props, { slots }) {
+    const context = useStyleProvider(props);
+    const state = reactive<Partial<StyleContextProps>>({
+      ...context.value,
+    });
+    provide(StyleContextKey, state);
```
[the remark this answers is not in the document]
Comment (CCherry07) [reply components/_util/cssinjs/StyleContext.tsx:145]: 要监听，props的变化，更新state，注入context, 如果使用useStyleProvider，监听useStyleProvider 返回值之后还要再更新注入，这样做的话就变得奇怪了，我觉得

[the remark this answers is not in the document]
Comment (CCherry07) [reply components/_util/cssinjs/StyleContext.tsx:145]:
```ts
provide(StyleContextKey, context);
```
这样写，会失去provide的响应式

```ts
export const useStyleProvider = (props: StyleProviderProps) => {
  const parentContext = useStyleInject();

  const context = computed<Partial<StyleContextProps>>(() => {
    const mergedContext: Partial<StyleContextProps> = {
      ...parentContext.value,
    };
    const propsValue = unref(props);
    Object.keys(propsValue).forEach(key => {
      const value = propsValue[key];
      if (propsValue[key] !== undefined) {
        mergedContext[key] = value;
      }
    });

    const { cache } = propsValue;
    mergedContext.cache = mergedContext.cache || createCache();
    mergedContext.defaultCache = !cache && parentContext.value.defaultCache;
    return mergedContext;
  });
  const state = reactive({ ...context.value });
  watch(context, (newValue) => {
    Object.keys(newValue).forEach(key => {
      state[key] = newValue[key];
    });
  });
  provide(StyleContextKey, state);
  return context;
};
```
这样写的话，useStyleProvider 每执行一次 都会重新执行provide

[the remark this answers is not in the document]
Comment (tangjinzhou) [reply components/_util/cssinjs/StyleContext.tsx:145]: 1、useStyleProvider 的参数 props，就是原始 props，本身就是响应式的呀
2、useStyleProvider 本身就只会执行一次啊，只在组件实例化的时候执行一次

**Review by tangjinzhou — APPROVED** [review]

**Comment by tangjinzhou** [comment]
我改了下，hashPriority 直接不支持动态得了，这个动态的意义不大

**Comment by CCherry07** [comment]
我看了一下这个 hashPriority 用了inject 是动态的，不过我也面的css 权重where 没有改，我前面看了antd 的，他们好像也不是能动态改的，我本来想，做一个forceRender的

**Comment by CCherry07** [comment]
还有一件事情，就是我们现在的antdv4,中的classname 会有hashid 这个，看不能给去掉，antd 是没有的

**Pull request merged**